src/Controller/AuthenticationController.php

```diff
@@ -373,12 +373,13 @@ private function sendNotification($notificationType, $notificationAddress)
             $notificationAddress
         ));
 
+        // The GCM API has been replaced by FCM
         if ($notificationType == 'GCM') {
```

Comment:
I don't recall the decision to 'upcycle' GCM to FCM in the Tiqr-GSSP. Maybe this was a construction to facilitate a rollover period where we had to support both protocols? Removing this BC support would improve the code as a whole in my opinion. As it removes unpredictable behavior.

And adding to that, I think this has become an unreachable condition. As configuring GCM has been made impossible by removing it from the `Configuration` parser.

Reply:
This is a leftover of the migration from GCM to FCM where we had many clients out that did not update their device notification address with the token exchange. So both the GCM and FCM APIs were used to try and send the push notification. These client have since been updated and Google has switched of the GCM API. The Tiqr Android clients still always report `GCM` as their notificationType, so this is what is stored in the UserStorage for all Android clients. 
The Tiqr_Message class to use must always be `FCM` however (Tiqr_Message_FCM).

Reply:
Thanks for the clarification!